src/Network/WebSockets/TSSocketManager.m

```diff
@@ -127,13 +146,28 @@ - (void)becomeActive
 }
 
 + (void)resignActivity {
-    SRWebSocket *socket = [[self sharedManager] websocket];
-    [socket close];
+    OWSAssert([NSThread isMainThread]);
```

Comment:
whoops. I see you've already addressed this in Signal-iOS:charlesmchen/fixWebsocket

:+1: